views-react/src/main/java/io/micronaut/views/react/ReactJSBeanFactory.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.views.react;
+
+import io.micronaut.context.ApplicationContext;
+import io.micronaut.context.annotation.Bean;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.io.ResourceResolver;
+import io.micronaut.views.react.util.BeanPool;
+import io.micronaut.views.react.util.JavaUtilLoggingToSLF4J;
+import io.micronaut.views.react.util.OutputStreamToSLF4J;
+import jakarta.inject.Named;
+import jakarta.inject.Singleton;
+import org.graalvm.polyglot.Engine;
+import org.graalvm.polyglot.HostAccess;
+import org.graalvm.polyglot.SandboxPolicy;
+import org.graalvm.polyglot.Source;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.event.Level;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import static java.lang.String.format;
+
+/**
+ * Allows the default Javascript context and host access policy to be controlled.
+ */
+@Factory
+@Internal
+class ReactJSBeanFactory {
+    private static final Logger LOG = LoggerFactory.getLogger("js");
+
+    // We cache the Source objects because they are expensive to create, but, we don't want them
+    // to be singleton beans so we can recreate them on file change.
+    private Source serverBundle;  // L(this)
```

Comment:
can we use a different scope, e.g. `@Refreshable` instead of this custom caching?

Reply:
Maybe, I'm not sure. It's triggered by a file watch event and has to be integrated with the BeanPool.